sdk/src/Services/DocDB/Custom/Internal/PreSignedUrlRequestHandler.cs

```diff
@@ -98,7 +97,7 @@ protected virtual void PreInvoke(IExecutionContext executionContext)
                 var iRequest = marshaller.Marshall(preSignedUrlRequest as AmazonWebServiceRequest);
                 iRequest.UseQueryString = true;
                 iRequest.HttpMethod = HTTPGet;
-                iRequest.Endpoint = new UriBuilder(UriSchemeHTTPS, endpoint.GetEndpointForService(config).Hostname).Uri;
+                iRequest.Endpoint = new Uri(config.DetermineServiceOperationEndpoint(new Runtime.Endpoints.ServiceOperationEndpointParameters(executionContext.RequestContext.OriginalRequest, RegionEndpoint.GetBySystemName(preSignedUrlRequest.SourceRegion))).URL);
```

Comment:
This is using the new logic I added for the endpoints 2.0 code path allowing to pass in an alternate endpoint. In that case the source region of the resource being copied. This same logic is done in other services that have a copy feature like RDS and Neptune.